### ISSUE TITLE
Directory paths are not cleaned if fn_clean_exts is used in config

### ISSUE THREAD

**Issue by pooranis** [ewels/MultiQC#897]
**Description of bug:**
[Gist of example log file for cutadapt.](https://gist.github.com/pooranis/cf2e332bc272c083dbad4755c347801f)
[cutadapt_example.log](https://github.com/ewels/MultiQC/files/2829552/cutadapt_example.log)

When I run multiqc with no special config options on the example log file, the filename gets cleaned to: `Blank10_1_L001_R2_001`.  I then tried running with the following config:

```yaml
fn_clean_exts:
    - type: regex
      pattern: '_\d+_L001_R[12]_001\.fastq\.gz'
```
This produces a cleaned filename with the extension removed, but the directory included: `/mnt/EFS/user_uploads/6ea525c94095/tmp/qiime2-archive-mvdsp9fc/a388e376-925d-4671-aa5e-8284ee2a0add/data/Blank10` which is not what I expected.

**MultiQC run details:**
 - Command used to run MultiQC: `multiqc -c ./multiqc_config.yaml ./dir_containing_cutadapt_example_log`
where *multiqc_config.yaml* just contains the config above
 - MultiQC Version: MultiQC 1.7
 - Operating System: Red Hat Enterprise Linux Server release 7.2
 - Python Version: 3.7.1
 - Method of MultiQC installation: pip

 **Additional context**
I would like to replace the [default fn_clean_exts](https://github.com/ewels/MultiQC/blob/master/multiqc/utils/config_defaults.yaml) because it sometimes removes strings I want to keep, and I would like to just remove 2-3 specific patterns.  Is there a way to do this while still allowing MultiQC to clean the directory path?

**Comment by pooranis** [comment]
I wonder if the problem is from [here in base_module.py](https://github.com/ewels/MultiQC/blob/master/multiqc/modules/base_module.py#L209)

For type 'truncate', the default, it does this:
```python
                if ext['type'] == 'truncate':
                      s_name = os.path.basename(s_name.split(ext['pattern'], 1)[0])
```
But, for 'regex'
```python
                elif ext['type'] == 'regex':
                    s_name = re.sub(ext['pattern'], '', s_name)
```
If we changed to `s_name = os.path.basename(re.sub(ext['pattern'], '', s_name))`, would that work (and not interfere with anything else in multiqc)?  If it doesn't interfere with any other options/configs, I can submit pull request to fix.

**Comment by ewels** [comment]
So, I just had a quick dig with this. I was a little confused as to why we would need this at all, as the default `f['s_name']` that comes from the `os.path.walk` function that finds all files returns base names already. Then I found an [ancient comment](https://github.com/ewels/MultiQC/pull/156#issuecomment-196840754) from @lpantano pointing out that some modules (cutadapt was the culprit there too) take their sample names from the file contents. In those specific cases, the sample names can be complete paths. She added the `truncate` basename function you pointed to for that reason, but at the time the other methods didn't exist. I guess the combination of those modules + long paths + specific cleaning _other_ than the default truncate is not very common which is why no-one has mentioned this since then.

Anyway, I have generalised this code a little and moved the `os.path.basename` call up in the code so that it's already applied for all cleaning methods. Hopefully this won't break anyone's systems (you [never know](https://xkcd.com/1172/)!), I think for the majority it won't change anything. For those who want file paths, there is always `-d`.

Many thanks for spotting this irregularity and reporting it! Sorry it took me so long to get to the issue.
